src/linter/root.go

```diff
@@ -810,6 +821,70 @@ func (d *RootWalker) enterClassMethod(meth *stmt.ClassMethod) bool {
 	return false
 }
 
+type classPhpDocParseResult struct {
+	properties meta.PropertiesMap
+}
+
+func (d *RootWalker) parsePHPDocClass(doc string) (classPhpDocParseResult, []string) {
+	var result classPhpDocParseResult
+	var phpDocErrors []string
+
+	if doc == "" {
+		return result, nil
+	}
+
+	result.properties = make(meta.PropertiesMap)
+
+	// TODO(quasilyte): reduce code duplication in various phpdoc parsing methods.
```

Comment:
I guess you could do it in this pull request as it should not be that hard and I think that I have started to do it already but I am not 100% sure about that.

Reply:
There exists phpdoc package and phpdoc.Parse() method. I am not sure why I didn't use it everywhere